### PR TITLE
Pricing UI - Authorization Checks + Polish 

### PR DESCRIPTION
**Pull request by ClaudeROSSI**
Authorization:
- Access to the Pricing Definition List is drived by the site.**canMaintainPricingDefinitions** property
- Check that PRICING Module is ON 

UI Look&Feel
- Connector Filter - new icons for selecting all connectors

Polish
- changes on the pipe to format prices and labels
- new regexp for validating amount inputs
- some renaming (e.g.: TableViewPricingsActionDef ==> TableView**PricingDefinitions**ActionDef)

### CODE REVIEW

File: src/app/pages/charging-stations/list/charging-stations-list-table-data-source.ts [hunk 1]
```diff
@@ -49,12 +49,13 @@ import { ChargingStationsConnectorsDetailComponent } from '../details-component/
...
 export class ChargingStationsListTableDataSource extends TableDataSource<ChargingStation> {
   private readonly isOrganizationComponentActive: boolean;
+  private readonly isPricingComponentActive: boolean;
   private editAction = new TableEditChargingStationAction().getActionDef();
   private smartChargingAction = new TableChargingStationsSmartChargingAction().getActionDef();
   private deleteAction = new TableDeleteChargingStationAction().getActionDef();
   private generateQrCodeConnectorAction = new TableChargingStationGenerateQrCodeConnectorAction().getActionDef();
   private canExport = new TableExportChargingStationsAction().getActionDef();
-  private viewPricingsAction = new TableViewPricingsAction().getActionDef();
+  private maintainPricingDefinitionsAction = new TableViewPricingDefinitionsAction().getActionDef();
```
Comment (LucasBrazi06) [inline src/app/pages/charging-stations/list/charging-stations-list-table-data-source.ts:58]: maintain (maintainPricingDefinitionsAction) <> view (TableViewPricingDefinitionsAction)
View is R/O, Maintain is R/W (should be Edit btw)

File: src/app/pages/organization/sites/list/sites-list-table-data-source.ts [hunk 2]
```diff
@@ -48,7 +51,7 @@ export class SitesListTableDataSource extends TableDataSource<Site> {
   private exportOCPPParamsAction = new TableExportOCPPParamsAction().getActionDef();
   private siteGenerateQrCodeConnectorAction = new TableSiteGenerateQrCodeConnectorAction().getActionDef();
   private createAction = new TableCreateSiteAction().getActionDef();
-  private viewPricingsAction = new TableViewPricingsAction().getActionDef();
+  private maintainPricingDefinitionsAction = new TableViewPricingDefinitionsAction().getActionDef();
```
Comment (LucasBrazi06) [inline src/app/pages/organization/sites/list/sites-list-table-data-source.ts:54]: Same comment

File: src/app/shared/pricing-definitions/pricing-definition/pricing-definition.component.ts [hunk 3]
```diff
@@ -97,23 +99,21 @@ export class PricingDefinitionComponent implements OnInit {
...
         Validators.compose([
           Validators.required,
-          Validators.maxLength(100),
         ])),
       staticRestrictions: new FormGroup({
         validFrom: new FormControl(null),
         validTo: new FormControl(null),
         connectorPowerEnabled: new FormControl(false),
-        connectorPowerkW: new FormControl(null, Validators.pattern('[0-9]*[,.]?[0-9]{1,2}')),
+        connectorPowerkW: new FormControl(null, Validators.pattern(/^-?((\d+(\.\d+)?))$/)),
```
Comment (LucasBrazi06) [inline src/app/shared/pricing-definitions/pricing-definition/pricing-definition.component.ts:116]: Could be useful to use a constant for the regex ;-)

File: src/app/shared/pricing-definitions/pricing-definitions-table-data-source.ts [hunk 4]
```diff
@@ -114,48 +117,28 @@ export class PricingDefinitionsTableDataSource extends DialogTableDataSource<Pri
       {
         id: 'dimensions.flatFee.price',
         name: 'settings.pricing.flat_fee',
-        formatter: (price: number) => {
-          if (price === undefined) {
-            return '-';
-          }
-          return this.appPricingDimensionsUnit.transform('flat_fee_formatted_price', price);
-        },
+        formatter: (price: number) => this.appPricingDimensionsPrice.transform('flat_fee_formatted_price', price),
```
Comment (LucasBrazi06) [inline src/app/shared/pricing-definitions/pricing-definitions-table-data-source.ts:120]: Much cleaner ;-)

File: src/app/shared/table/actions/table-view-pricing-definitions-action.ts [hunk 5]
```diff
@@ -8,20 +8,20 @@ import { PricingButtonAction, PricingDefinitionDialogData } from '../../../types
...
 import { TableViewPricingListAction } from './table-pricing-action';
 
-export interface TableViewPricingsActionDef extends TableActionDef {
+export interface TableViewPricingDefinitionsActionDef extends TableActionDef {
   action: (pricingDialogComponent: ComponentType<unknown>, dialog: MatDialog,
     dialogParams: DialogParams<PricingDefinitionDialogData>, refresh?: () => Observable<void>) => void;
 }
-export class TableViewPricingsAction extends TableViewPricingListAction {
-  public getActionDef(): TableViewPricingsActionDef {
+export class TableViewPricingDefinitionsAction extends TableViewPricingListAction {
```
Comment (LucasBrazi06) [inline src/app/shared/table/actions/table-view-pricing-definitions-action.ts:15]: Should extend TableViewAction?
Missing Edit action that extends TableEditAction?

File: src/app/types/Authorization.ts [hunk 6]
```diff
@@ -182,6 +182,7 @@ export interface SiteAuthorizationActions extends AuthorizationActions {
   canReadUsers?: boolean;
   canExportOCPPParams?: boolean;
   canGenerateQrCode?: boolean;
+  canMaintainPricingDefinitions?: boolean;
```
Comment (LucasBrazi06) [inline src/app/types/Authorization.ts:185]: canEditPricingDefinitions?
canViewPricingDefinitions?

File: src/assets/i18n/en.json [hunk 7]
```diff
@@ -33,7 +33,6 @@
     "unexpected_error_backend": "An unexpected error occurred, check the logs",
     "select_at_least_one_record": "You must select at least one record in the list",
     "mandatory_field": "Required field",
-    "only_number_field": "Only numbers are allowed",
```
Comment (LucasBrazi06) [inline src/assets/i18n/en.json:36]: WL (Wrong Line)

**Review by LucasBrazi06 — APPROVED** [review]
Check for the icing

File: src/app/shared/pricing-definitions/pricing-definition/pricing-definition.component.ts (hunk 3 above)
Comment (cabch) [reply src/app/shared/pricing-definitions/pricing-definition/pricing-definition.component.ts:116]: will push the change ✅

File: src/assets/i18n/en.json (hunk 7 above)
Comment (cabch) [reply src/assets/i18n/en.json:36]: I deleted it as we already have the 'error_number_pattern' key, not sure why you say WL here

**Pull request merged**